MODULE.bazel

```diff
@@ -2,8 +2,31 @@
 
 module(
     name = "com_myorg_rules_mylang",
-    version = "0.0.0",
-    compatibility_level = 1,
+
+    # NOTE:
+    # version = "",
+    #
+    # Always leave version unset or set to "" (the default). The default value
+    # can prevent issues when the module is used via non-registry overrides
+    # (e.g. https://github.com/bazel-contrib/rules_go/issues/4380).
+    #
+    # The publish.yaml GitHub Action sets the version in the registry to the
+    # release version by patching this MODULE.bazel file in the pull request to
+    # the BCR.
+    #
+    # For more info, see this Slack thread:
+    # https://bazelbuild.slack.com/archives/CA31HN1T3/p1750406404452179
+
+    # NOTE:
+    # compatibility_level = 0,
+    #
+    # The compatibility_level should only be bumped when an incompatible
```

Comment:
Let's stress that this change should be very breaking (that is, affects most use cases and isn't easy to migrate past), otherwise bumping compatibility_level will be more of a hassle.

Cc @Wyverald

Reply:
Done, let me know if the new wording is better. Happy to change it again if needed!